# An illustration of "smoothing via adaptive shrinkage" applied to
# chromatin immunoprecipitation sequencing ("ChIP-seq") data. Compare
# the plot generated by the code below against the figure in the
# "ChIP-seq data" section of the paper.
#
# The ChIP-seq data are sequencing read counts for transcription
# factor YY1 in cell line GM12878, restricted to 880,001--1,011,072 bp
# on chromosome 1. These data were collected as part of the ENCODE
# ("Encyclopedia Of DNA Elements") project.
#
library(smashr)
library(scales)

# LOAD DATA
# ---------
# Load the ChIP-seq data.
cat("Loading Chip-seq data.\n")
load("../data/reg_880000_1011072.RData")

# PREPARE DATA
# ------------
# Prepare the data for analysis with smash.
bppos = 880001:1011072
peaks=read.table("../data/Gm1287peaks_chr1_sorted.txt")
peaks=peaks[order(peaks[,2]),]
all.base=1:peaks[dim(peaks)[1],3]
peaks.start=peaks[,2][peaks[,2]>=880000&peaks[,2]<=1011072]
peaks.end=peaks[,3][peaks[,3]>=880000&peaks[,3]<=1011072]

# RUN SMASH
# ---------
# Apply smash to the Motorcycle Accident data set.
cat("Running smash on ChIP-seq data.\n")
res = smash.poiss(M[1,]+M[2,],post.var=TRUE)

# PLOT RESULTS
# ------------
# The first plot shows counts (summed across two replicate
# experiments), with darker areas corresponding to higher numbers of
# data points.
plot(bppos,M[1,]+M[2,],xlab="position",ylab="counts",pch=16,cex=0.5,
     col=alpha("black",0.04))
invisible(readline(prompt = "Press [enter] to continue analysis... "))

# The second plot shows the estimated intensity function from smash
# (black line) and the location of the peaks called by MACS (red
# markers beneath the estimated intensity).
plot(bppos,res$est,type='l',xlab="position",ylab="intensity")
for(i in 1:length(peaks.start))
  segments(peaks.start[i],-0.02,peaks.end[i],-0.02,col=2,lwd=4)


