---
title: "ChIP-seq example"
author: "Zhengrong Xing, Peter Carbonetto and Matthew Stephens"
output: workflowr::wflow_html
---

An illustration of "smoothing via adaptive shrinkage" (SMASH) applied
to chromatin immunoprecipitation sequencing ("ChIP-seq") data. This
implements the SMASH analysis presented in Sec. 5.2 of the manuscript.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,fig.align = "center")
```

Set up environment
------------------

We begin by loading the smashr, ggplot2 and cowplot packages, as well
as some additional functions used to implement the analysis below.

```{r load-pkgs, warning=FALSE}
source("../code/chipseq.functions.R")
library(smashr)
library(ggplot2)
library(cowplot)
```

Load the ChIP-seq data
----------------------

The ChIP-seq data are sequencing read counts for transcription
factor YY1 in cell line GM12878, restricted to 880,001--1,011,072 bp
on chromosome 1. These data were collected as part of the ENCODE
("Encyclopedia Of DNA Elements") project.

```{r load-data}
load("../data/reg_880000_1011072.RData")
bppos  <- 880001:1011072
counts <- M[1,] + M[,2]
```

Note that there are two replicates of the GM12878 cell line, so we
analyze the combined read counts from both replicates, stored in the
`counts` vectors.

Run SMASH
---------

Next, we apply SMASH to the read counts to estimate the mean and
variance of the underlying signal. It may take several minutes to
complete this step.

```{r run-smash}
res <- smash.poiss(counts,post.var = TRUE)
```

Plot the SMASH estimates
------------------------

To provide a "baseline" to compare against the SMASH estimates, we
retrieve the peaks identified in the same ChIP-seq data using the MACS
software.

```{r load-macs-peaks}
macs.file <- "../data/Gm1287peaks_chr1_sorted.txt"
peaks <- read.macs.peaks(macs.file,min(bppos),max(bppos))
```

TO DO: Describe what is being shown in the plot.

```{r plot-smash-estimates, fig.height=4, fig.width=8}
create.chipseq.plot(bppos/1e6,counts,res$est,(peaks$start + peaks$end)/2e6,
                    nbreaks = 10) +
  scale_x_continuous(limits = c(0.88,1.02),breaks = seq(0.88,1.02,0.02)) +
  scale_y_continuous(limits = c(-1,9),breaks = seq(0,8,2))
```
